cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DashboardHttpHandler.java

```diff
@@ -92,48 +106,50 @@ public synchronized void list(final HttpRequest request, final HttpResponder res
 
   @Path("/{dashboard-id}")
   @DELETE
-  public synchronized void delete(final HttpRequest request, final HttpResponder responder,
+  public void delete(final HttpRequest request, final HttpResponder responder,
                                   @PathParam("namespace-id") String namespace,
                                   @PathParam("dashboard-id") String id) throws Exception {
-    if (!configStore.contains(namespace, id)) {
-      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
-    } else {
-      configStore.remove(namespace, id);
+    try {
+      configStore.delete(namespace, CONFIG_TYPE, id);
       responder.sendStatus(HttpResponseStatus.OK);
+    } catch (ConfigNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Dashboard not found");
     }
   }
 
   @Path("/{dashboard-id}")
   @GET
-  public synchronized void get(final HttpRequest request, final HttpResponder responder,
+  public void get(final HttpRequest request, final HttpResponder responder,
                                @PathParam("namespace-id") String namespace,
                                @PathParam("dashboard-id") String id) throws Exception {
-    if (!configStore.contains(namespace, id)) {
-      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
-    } else {
+    try {
+      Config dashConfig = configStore.get(namespace, CONFIG_TYPE, id);
       JsonObject jsonObject = new JsonObject();
-      jsonObject.addProperty("id", id);
-      jsonObject.add("config", new JsonParser().parse(configStore.get(namespace, id)));
+      jsonObject.addProperty(ID, id);
+      jsonObject.add(PROPERTY_NAME, new JsonParser().parse(dashConfig.getProperties().get(PROPERTY_NAME)));
```

Comment:
add a comment that you unwrap whatever is inside to become a normal JSON object in the output, not a String. Will help read the code. Also, may be even add a comment somewhere which describes the Config layout for dashboard configuration. May be near the ConfigStore configStore field or where you create dashboard config